packages/react-charting/UnitTests/GroupedVerticalBarChartUT.test.tsx

```diff
@@ -0,0 +1,524 @@
+    const instance = new GroupedVerticalBarChartBase({
+      data: chartPoints,
+    });
+    expect(instance).toBeDefined();
+    const dataSet = instance._createDataSetOfGVBC(chartPoints);
+    expect(dataSet).toBeDefined();
+    expect(dataSet['keys']).toHaveLength(2);
+    expect(dataSet['xAxisLabels']).toHaveLength(4);
+    instance._getMargins(margins);
+    instance._keys = dataSet['keys'];
+    instance._xAxisLabels = dataSet['xAxisLabels'];
+    instance._classNames = getClassNames(instance.props.styles!, {
+      theme: instance.props.theme!,
+      href: instance.props.href!,
+    });
+    var result = instance._getDomainMargins(50);
+    expect(result['bottom']).toEqual(10);
+    expect(result['left']).toEqual(18);
+    expect(result['right']).toEqual(18);
+    expect(result['top']).toEqual(10);
+  });
+});
+
+runTest('Get Scales', () => {
+  test('Should return correct x0Scale and x1Scale', () => {
+    const instance = new GroupedVerticalBarChartBase({
+      data: chartPoints,
+    });
+    expect(instance).toBeDefined();
+    const dataSet = instance._createDataSetOfGVBC(chartPoints);
+    expect(dataSet).toBeDefined();
+    expect(dataSet['keys']).toHaveLength(2);
+    expect(dataSet['xAxisLabels']).toHaveLength(4);
+    instance._getMargins(margins);
+    instance._keys = dataSet['keys'];
+    instance._xAxisLabels = dataSet['xAxisLabels'];
+    instance._classNames = getClassNames(instance.props.styles!, {
+      theme: instance.props.theme!,
+      href: instance.props.href!,
+    });
+    var x0Scale = instance._createX0Scale(1000);
+    expect(x0Scale).not.toBeNull();
+    const domains = x0Scale['domain'];
+    const ranges = x0Scale['range'];
+    expect(domains).not.toBeNull();
+    expect(ranges).not.toBeNull();
+    var x1Scale = instance._createX1Scale();
+    expect(x1Scale).not.toBeNull();
+  });
+});
+
+runTest('_buildGraph', () => {
+  test('Should return the correct graph data', () => {
+    const instance = new GroupedVerticalBarChartBase({
+      data: chartPoints,
+    });
+    expect(instance).toBeDefined();
+    const dataSet = instance._createDataSetOfGVBC(chartPoints);
+    expect(dataSet).toBeDefined();
+    expect(dataSet['keys']).toHaveLength(2);
+    expect(dataSet['xAxisLabels']).toHaveLength(4);
+    instance._getMargins(margins);
+    instance._keys = dataSet['keys'];
+    instance._xAxisLabels = dataSet['xAxisLabels'];
+    instance._datasetForBars = datasetForBars;
+    instance._classNames = getClassNames(instance.props.styles!, {
+      theme: instance.props.theme!,
+      href: instance.props.href!,
+    });
+    var x0Scale = instance._createX0Scale(1000);
+    expect(x0Scale).not.toBeNull();
+    const domains = x0Scale['domain'];
+    const ranges = x0Scale['range'];
+    expect(domains).not.toBeNull();
+    expect(ranges).not.toBeNull();
+
+    var x1Scale = instance._createX1Scale();
+    expect(x1Scale).not.toBeNull();
```

Comment:
why duplicate testing

Reply:
We are using domains and ranges values in this test so added extra check here